diff --git a/python/paddle/fluid/tests/unittests/op_test.py b/python/paddle/fluid/tests/unittests/op_test.py
index a3e1650c..c9c3f0e9 100644
--- a/python/paddle/fluid/tests/unittests/op_test.py
+++ b/python/paddle/fluid/tests/unittests/op_test.py
@@ -1350,21 +1350,10 @@ class OpTest(unittest.TestCase):
                     ")")
 
     def _get_places(self):
-        if self.dtype == np.float16:
-            if core.is_compiled_with_cuda() and core.op_support_gpu(
-                    self.op_type):
-                place = core.CUDAPlace(0)
-                if core.is_float16_supported(place):
-                    return [place]
-                else:
-                    return []
-            else:
-                return []
-        places = [fluid.CPUPlace()]
-        cpu_only = self._cpu_only if hasattr(self, '_cpu_only') else False
-        if core.is_compiled_with_cuda() and core.op_support_gpu(self.op_type)\
-           and not cpu_only:
-            places.append(core.CUDAPlace(0))
+        all_pluggable_device_type = core.list_all_pluggable_device_type()
+        places = []
+        for device_type in all_pluggable_device_type:
+            places.append(core.PluggableDevicePlace(device_type, 0))
         return places
 
     def check_output(self,
